### PR TITLE
TransposeSinking refactoring: part 2 (class names, folders, file names)

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
All TransposeSinking transformations and tests were moved to transformations/transpose_sinking folder
Introduced ov::pass::transpose_sinking namespace
Transformation classes were renamed: TransposeSinking* -> TS* 
Transformation files were renamed: transpose_sinking_* -> ts_*

  before: ______________________________________________after: _______________________________
image image

### CODE REVIEW

**Review by evkotov — APPROVED** [review]

File: src/frontends/tensorflow_lite/src/frontend.cpp [hunk 1]
```diff
@@ -268,7 +268,7 @@ void FrontEnd::normalize(const std::shared_ptr<ov::Model>& function) const {
     manager.register_pass<ov::frontend::tensorflow_lite::pass::TFLQuantizeResolver>();
     manager.register_pass<ov::frontend::tensorflow_lite::pass::Rfft2dSimplifier>();
     manager.register_pass<ov::pass::TransposeSinking>();
-    manager.register_pass<ov::pass::TransposeSinkingGeneral>();
+    manager.register_pass<ov::pass::transpose_sinking::TSGeneral>();
```
Comment (rkazants) [inline src/frontends/tensorflow_lite/src/frontend.cpp:271]: To me `ov::pass::TransposeSinking` looks better. You can include all particular TS passes into `transpose_sinking` namespace and remain the main entry on `pass` namespace.

**Review by rkazants — COMMENTED** [review]
Left one comment about naming and using additional namespace only for particular TS passes.

Other looks good to me

File: src/frontends/tensorflow_lite/src/frontend.cpp (hunk 1 above)
Comment (itikhono) [reply src/frontends/tensorflow_lite/src/frontend.cpp:271]: unfortunately, we already have ov::pass::TransposeSinking (old common transformation), see line 270
We will be able to replace it in the next Release

Comment (rkazants) [reply src/frontends/tensorflow_lite/src/frontend.cpp:271]: I see what about `TransposeSinkingExtended` or `TransposeSinkingAdvanced` names?

Comment (itikhono) [reply src/frontends/tensorflow_lite/src/frontend.cpp:271]: I believe we can leave both options:
ov::pass::TransposeSinkingGeneral (the alias)
ov::pass::transpose_sinking::TSGeneral

**Review by rkazants — APPROVED** [review]

File: src/frontends/tensorflow_lite/src/frontend.cpp (hunk 1 above)
Comment (itikhono) [reply src/frontends/tensorflow_lite/src/frontend.cpp:271]: it will be fixed in the next PR

**Pull request merged**